app/models/warehouse/build_report.rb

```diff
@@ -4,6 +4,32 @@ def run!
       return nil unless Warehouse::Base.enabled?
       fill_cas_report_table!
       fill_cas_non_hmis_client_history_table!
+      fill_cas_vacancy_table!
+    end
+
+    def fill_cas_vacancy_table!
+      Warehouse::CasVacancy.delete_all
+
+      ::Voucher.all.each do |voucher|
+        vacancy = Warehouse::CasVacancy.new(program_id: voucher.sub_program.program.id, sub_program_id: voucher.sub_program_id)
+
+        if voucher.deleted_at
+          vacancy.destroy!
```

Comment:
This shouldn't be necessary.  `vacancy` hasn't been persisted at this time and you only call `save!` in the else clause.  Perhaps, just `if voucher.deleted_at.blank?`